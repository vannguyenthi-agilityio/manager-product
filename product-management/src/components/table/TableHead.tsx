// Chakra ui libs
import { Flex, Thead, Tr, Th, Input, Box, TableHeadProps as TableHeadPropsChakra } from '@chakra-ui/react';

// Components
import { Select } from '@components/common/Select';

// Types
import { ColumnHeader } from '@types';

// Utils
import { convertStringToCapitalize } from '@utils/index';

// Constants
import { PRODUCT_STATUS, PRODUCT_TYPE } from '@constants';

interface TableHeadProps extends TableHeadPropsChakra {
  columns?: ColumnHeader[];
  onSearchClick: (value: ColumnHeader) => void;
}

export const TableHead = ({ columns = [], onSearchClick, ...props }: TableHeadProps) => {
  const status = Object.values(PRODUCT_STATUS).map((status) => ({
    label: convertStringToCapitalize(status === PRODUCT_STATUS.SOLD_OUT ? 'Sold Out' : status),
    value: status
  }));

  const types = Object.values(PRODUCT_TYPE).map((type) => ({
    label: convertStringToCapitalize(type),
    value: type
  }));
  return (
    <Thead
      {...props}
      w='full'
    >
      <Tr>
        {columns.map((column) => (
          <Th
            key={column.value}
            cursor={column.filterable ? 'pointer' : 'default'}
            display={column.value === 'Action' ? 'flex' : 'revert'}
            justifyContent={column.value === 'Action' ? 'center' : 'left'}
          >
            <Flex
              flexDirection='column'
              onClick={() => onSearchClick(column)}
            >
              {column.value}
              {column.value === 'Type' || column.value === 'Status' ? (
                <Box w='60px'>
                  <Select
                    options={column.value === 'Type' ? types : status}
                    variant='unstyled'
                    placeholder='All'
                    size='xs'
                  />
                </Box>
              ) : (
                column.value !== 'Action' && (
                  <Input
                    placeholder='Search'
                    size='xs'
                    variant='unstyled'
                  />
                )
              )}
            </Flex>
          </Th>
        ))}
      </Tr>
    </Thead>
  );
};
