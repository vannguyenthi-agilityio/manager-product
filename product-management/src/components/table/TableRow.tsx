// Chakra ui libs
import {
  Text,
  Flex,
  Tbody,
  Tr,
  Td,
  Button,
  Menu,
  MenuList,
  MenuButton,
  MenuItem,
  TableCaption,
  TableBodyProps as TableBodyPropsChakra
} from '@chakra-ui/react';

// Types
import { Product } from '@types';

// Constans
import { PRODUCT_STATUS } from '@constants';

// Utils
import { convertPxToRem, filterItemInArray, convertStringToCapitalize, formatCurrencyUSD } from '@utils/index';

// Constants
import { COLORS } from '@constants';

interface TableRowProps extends TableBodyPropsChakra {
  data: Product[];
  filteredItems?: string[];
  onActionProduct: (product: Product, action: string) => void;
}

const TableRow = ({ data = [], filteredItems = [], onActionProduct, ...props }: TableRowProps) => {
  const handleEditProduct = (product: Product) => {
    //TODO: will handle to open Edit Product modal
    onActionProduct(product, 'edit');
  };

  const handleDeleteProduct = (product: Product) => {
    //TODO: will handle to open Confirm Delete Product modal
    onActionProduct(product, 'delete');
  };

  const renderTableTd = (key: string, value: string) => {
    switch (key) {
      case 'name':
        return (
          <Text
            fontSize='sm'
            fontWeight='normal'
          >
            {convertStringToCapitalize(value)}
          </Text>
        );
      case 'status':
        return (
          <Button
            size='xs'
            colorScheme={convertStringToCapitalize(value) === 'Available' ? 'mania' : 'red'}
          >
            {convertStringToCapitalize(value === PRODUCT_STATUS.SOLD_OUT ? 'Sold Out' : value)}
          </Button>
        );
      case 'quantity':
        return (
          <Text
            fontSize='sm'
            fontWeight='normal'
            style={{ border: '1px solid #dfe2e9', borderRadius: '8px', textAlign: 'center', width: '56px' }}
          >
            {value}
          </Text>
        );
      case 'price':
        return (
          <Text
            fontSize='sm'
            fontWeight='normal'
          >
            {formatCurrencyUSD.format(Number(value))}
          </Text>
        );
      default:
        return (
          <Text
            fontSize='sm'
            fontWeight='normal'
            textAlign={{ sm: 'center', md: 'left' }}
          >
            {convertStringToCapitalize(value)}
          </Text>
        );
    }
  };

  return data.length > 0 ? (
    <Tbody {...props}>
      {data.map((item: Product) => {
        return (
          <Tr key={item.id}>
            {filterItemInArray(Object.keys(item), filteredItems).map((key: string) => {
              const value = item[key as keyof Product] as string;
              return <Td key={`${key}-${item.id}`}>{renderTableTd(key, value)}</Td>;
            })}
            <Td>
              <Flex justifyContent='center'>
                <Menu>
                  <MenuButton>
                    <Flex
                      w={convertPxToRem(50)}
                      height={convertPxToRem(30)}
                      justifyContent='center'
                      alignItems='center'
                    >
                      <Text
                        fontSize='xl'
                        bg={`${COLORS.GREY}.650`}
                        borderRadius='full'
                        h={convertPxToRem(4)}
                        w={convertPxToRem(4)}
                        border={`1px solid ${COLORS.GREY}.650`}
                      />
                      <Text
                        fontSize='xl'
                        bg={`${COLORS.GREY}.650`}
                        borderRadius='full'
                        h={convertPxToRem(4)}
                        w={convertPxToRem(4)}
                        mx={convertPxToRem(2)}
                        border={`1px solid ${COLORS.GREY}.650`}
                      />
                      <Text
                        fontSize='xl'
                        bg={`${COLORS.GREY}.650`}
                        borderRadius='full'
                        h={convertPxToRem(4)}
                        w={convertPxToRem(4)}
                        border={`1px solid ${COLORS.GREY}.650`}
                      />
                    </Flex>
                  </MenuButton>
                  <MenuList>
                    <MenuItem onClick={() => handleEditProduct(item)}>Edit</MenuItem>
                    <MenuItem
                      color='red'
                      onClick={() => handleDeleteProduct(item)}
                    >
                      {' '}
                      Delete
                    </MenuItem>
                  </MenuList>
                </Menu>
              </Flex>
            </Td>
          </Tr>
        );
      })}
    </Tbody>
  ) : (
    <TableCaption>Search not found</TableCaption>
  );
};

export default TableRow;
