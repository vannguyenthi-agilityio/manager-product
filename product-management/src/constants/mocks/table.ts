// Types
import { Product } from '@types';

// Constants
import { PRODUCT_STATUS, PRODUCT_TYPE } from '@constants';

const productColumns = [
  {
    value: 'Name',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Status',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Types',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Quantity',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Brand',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Price',
    filterable: {
      isFilter: false
    }
  },
  {
    value: 'Action',
    filterable: null
  }
];

const productData = [
  {
    id: '1',
    name: 'Louis Vuitton',
    status: PRODUCT_STATUS.AVAILABLE,
    types: PRODUCT_TYPE.RING,
    quantity: 9177,
    price: 452.85,
    brand: 'Evan Flores'
  },
  {
    id: '2',
    name: 'Johnson & Johnson',
    status: PRODUCT_STATUS.SOLD_OUT,
    types: PRODUCT_TYPE.DIAMOND,
    quantity: 3064,
    price: 901.35,
    brand: 'Arlene Wilson'
  },
  {
    id: '3',
    name: 'Starbuks',
    status: PRODUCT_STATUS.SOLD_OUT,
    types: PRODUCT_TYPE.GEMSTONE,
    quantity: 9195,
    price: 641.2,
    brand: 'Jennie Cooper'
  },
  {
    id: '4',
    name: 'The Walt Disney',
    status: PRODUCT_STATUS.AVAILABLE,
    types: PRODUCT_TYPE.METAL,
    quantity: 3128,
    price: 510.3,
    brand: 'Philip Steward'
  },
  {
    id: '5',
    name: 'Mitsubishi',
    status: PRODUCT_STATUS.AVAILABLE,
    types: PRODUCT_TYPE.RING,
    quantity: 9892,
    price: 510.3,
    brand: 'Jorge Black'
  },
  {
    id: '6',
    name: 'IBM',
    status: PRODUCT_STATUS.SOLD_OUT,
    types: PRODUCT_TYPE.GOLD,
    quantity: 9011,
    price: 845.59,
    brand: 'Gladys Jones'
  }
] as Product[];

export { productColumns, productData };
