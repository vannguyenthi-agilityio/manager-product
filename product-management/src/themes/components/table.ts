// Utils
import { convertPxToRem } from '@utils/common';

// Constants
import { COLORS } from '@constants';

const tableTheme = {
  baseStyle: {
    th: {
      textTransform: 'none'
    }
  },

  sizes: {
    default: {
      th: {
        fontSize: 'sm',
        lineHeight: 'extraSmall',
        px: 'tableSpacing.tighter',
        fontWeight: '600'
      },
      td: {
        px: 'tableSpacing.tighter',
        fontSize: 'sm',
        lineHeight: 'base',
        fontWeight: '600'
      }
    }
  },

  variants: {
    default: {
      table: {
        bg: `${COLORS.WHITE}`
      },
      th: {
        color: `${COLORS.BLACK}.500`,
        p: convertPxToRem(10)
      },
      thead: {
        width: 'full',
        tr: {
          px: convertPxToRem(20),
          h: convertPxToRem(40),
          borderBottomWidth: '1px',
          borderColor: `${COLORS.GREY}.600`,
          th: {
            '&:first-of-type': {
              borderLeftRadius: convertPxToRem(8)
            },
            '&:last-of-type': {
              borderRightRadius: convertPxToRem(8)
            }
          }
        }
      },
      tbody: {
        tr: {
          h: convertPxToRem(80),
          borderBottomWidth: '1px',
          borderColor: `${COLORS.GREY}.600`,
          cursor: 'pointer',
          _hover: {
            bg: `${COLORS.GREY}.50`,
            td: {
              '&:first-of-type': {
                borderLeftRadius: convertPxToRem(8)
              },
              '&:last-of-type': {
                borderRightRadius: convertPxToRem(8)
              }
            }
          }
        }
      }
    }
  }
};

export default tableTheme;
