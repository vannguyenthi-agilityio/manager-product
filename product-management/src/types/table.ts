// eslint-disable-next-line @typescript-eslint/no-unused-vars
interface Customizable<key> {
  [key: string]: boolean;
}

export interface ColumnHeader {
  value: string;
  filterable?: Customizable<'isFilter'> | null;
}
