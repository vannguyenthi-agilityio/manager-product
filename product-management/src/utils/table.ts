// Types
import { Product } from '@types';

export const filterItemInArray = (data: string[], items: string[]) =>
  data.filter((item: string) => ![...items].includes(item));

export const formatProductResponse = (productList: Product[]) =>
  productList?.map((product) => ({
    id: product.id,
    name: product.name,
    status: product.status,
    types: product.types,
    quantity: product.quantity,
    brand: product.brand,
    price: product.price
  }));
