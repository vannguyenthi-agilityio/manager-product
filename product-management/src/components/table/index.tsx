// Chakra ui libs
import { Table as TableChakra, TableProps as TablePropsChakra } from '@chakra-ui/react';

// Components
import { TableHead } from './TableHead';
import { TableRow } from './TableRow';

// Types
import { Product, ColumnHeader } from '@types';

interface TableProps extends TablePropsChakra {
  columns?: ColumnHeader[];
  data?: Product[];
  filteredItems?: string[];
  size?: 'default';
  variant?: 'default';
  className?: string;
  onSearchClick: (value: ColumnHeader) => void;
}

const Table = ({
  columns = [],
  data = [],
  filteredItems = [],
  size = 'default',
  variant = 'default',
  className = '',
  onSearchClick,
  ...props
}: TableProps) => {
  const handleSearchClick = (value: ColumnHeader) => {
    onSearchClick(value);
  };

  return (
    <TableChakra
      size={size}
      variant={variant}
      className={className}
      data-testid='product-table'
      {...props}
    >
      <TableHead
        columns={columns}
        onSearchClick={handleSearchClick}
      />
      <TableRow
        data={data}
        filteredItems={filteredItems}
      />
    </TableChakra>
  );
};

export default Table;
