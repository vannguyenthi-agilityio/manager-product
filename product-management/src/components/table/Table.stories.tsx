import Storybook from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Table from './index';

// Utils
import { formatProductResponse } from '@utils/table';

// Mocks
import { productColumns, productData } from '@constants/mocks/table';

export default {
  title: 'Components/Table',
  component: Table,
  argTypes: {
    backgroundColor: { control: 'color' }
  }
};

const formatProductData = formatProductResponse(productData);

const Template: Storybook.StoryFn<typeof Table> = (args) => {
  return (
    <Table
      {...args}
      onActionProduct={action('onActionProduct')}
    />
  );
};

export const ProductTable = Template.bind({});
ProductTable.args = {
  columns: productColumns,
  data: formatProductData,
  filteredItems: ['id']
};
