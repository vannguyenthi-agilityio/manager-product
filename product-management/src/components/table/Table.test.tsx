import { cleanup, render, fireEvent } from '@testing-library/react';

// Utils
import { formatProductResponse } from '@utils/table';

// Mocks
import { productColumns, productData } from '@constants/mocks/table';

// Types
import { Product } from '@types';

// Components
import Table from '.';

const formatProductData = formatProductResponse(productData) as Product[];

describe('Table render', () => {
  afterEach(cleanup);

  it('should render Table component', () => {
    const mockFunction = jest.fn();
    const { getByTestId } = render(
      <Table
        columns={productColumns}
        data={formatProductData}
        onSearchClick={mockFunction}
      />
    );

    const table = getByTestId('product-table');

    expect(table).toBeTruthy();
    expect(table).toMatchSnapshot();
  });

  it('should simulate click search and expect mock function to be called', () => {
    const mockFunction = jest.fn();
    const { getByText } = render(
      <Table
        columns={productColumns}
        data={formatProductData}
        onSearchClick={mockFunction}
      />
    );
    const searchColum = getByText('Name');

    fireEvent.click(searchColum);
    expect(mockFunction).toHaveBeenCalled();
  });
});
