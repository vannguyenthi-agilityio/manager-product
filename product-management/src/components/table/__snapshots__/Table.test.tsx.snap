// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render should render Table component 1`] = `
<table
  class="chakra-table css-0"
  data-testid="product-table"
>
  <thead
    class="css-13rbvqk"
  >
    <tr
      class="css-0"
    >
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Name
          <input
            class="chakra-input css-0"
            placeholder="Search"
          />
        </div>
      </th>
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Status
          <div
            class="css-q2ms4q"
          >
            <label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  data-testid="select"
                >
                  <option
                    value=""
                  >
                    All
                  </option>
                  <option
                    value="available"
                  >
                    Available
                  </option>
                  <option
                    value="sold_out"
                  >
                    Sold out
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </label>
          </div>
        </div>
      </th>
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Type
          <div
            class="css-q2ms4q"
          >
            <label>
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                  data-testid="select"
                >
                  <option
                    value=""
                  >
                    All
                  </option>
                  <option
                    value="Gemstone"
                  >
                    Gemstone
                  </option>
                  <option
                    value="metal"
                  >
                    Metal
                  </option>
                  <option
                    value="diamond"
                  >
                    Diamond
                  </option>
                  <option
                    value="gold"
                  >
                    Gold
                  </option>
                  <option
                    value="ring"
                  >
                    Ring
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </label>
          </div>
        </div>
      </th>
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Quantily
          <input
            class="chakra-input css-0"
            placeholder="Search"
          />
        </div>
      </th>
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Brand
          <input
            class="chakra-input css-0"
            placeholder="Search"
          />
        </div>
      </th>
      <th
        class="css-rm0brf"
      >
        <div
          class="css-j7qwjs"
        >
          Price
          <input
            class="chakra-input css-0"
            placeholder="Search"
          />
        </div>
      </th>
      <th
        class="css-vh7bht"
      >
        <div
          class="css-j7qwjs"
        >
          Action
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    class="css-0"
  >
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          1
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Louis vuitton
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Available
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Ring
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          9177
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Evan flores
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $452.85
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:r1:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:r1:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:r1:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:r1:-menuitem-:r2:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:r1:-menuitem-:r3:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          2
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Johnson & johnson
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Sold out
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Diamond
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          3064
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Arlene wilson
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $901.35
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:r5:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:r5:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:r5:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:r5:-menuitem-:r6:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:r5:-menuitem-:r7:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          3
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Starbuks
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Sold out
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Gemstone
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          9195
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Jennie cooper
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $641.20
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:r9:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:r9:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:r9:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:r9:-menuitem-:ra:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:r9:-menuitem-:rb:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          4
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          The walt disney
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Available
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Metal
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          3128
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Philip steward
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $510.30
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:rd:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:rd:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:rd:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:rd:-menuitem-:re:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:rd:-menuitem-:rf:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          5
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Mitsubishi
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Available
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Ring
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          9892
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Jorge black
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $510.30
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:rh:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:rh:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:rh:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:rh:-menuitem-:ri:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:rh:-menuitem-:rj:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="css-0"
    >
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          6
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Ibm
        </p>
      </td>
      <td
        class="css-0"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Sold out
        </button>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Gold
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
          style="border: 1px solid #dfe2e9; border-radius: 8px; text-align: center; width: 56px;"
        >
          9011
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          Gladys jones
        </p>
      </td>
      <td
        class="css-0"
      >
        <p
          class="chakra-text css-m50rwi"
        >
          $845.59
        </p>
      </td>
      <td
        class="css-0"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            aria-controls="menu-list-:rl:"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-menu__menu-button css-1eqi7qs"
            id="menu-button-:rl:"
          >
            <span
              class="css-xl71ch"
            >
              <div
                class="css-1u4cqe9"
              >
                <p
                  class="chakra-text css-1vtejss"
                />
                <p
                  class="chakra-text css-18vkfhs"
                />
                <p
                  class="chakra-text css-1vtejss"
                />
              </div>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-sjt5nk"
              id="menu-list-:rl:"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-:rl:-menuitem-:rm:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                Edit
              </button>
              <button
                class="chakra-menu__menuitem css-1jf287i"
                data-index="1"
                id="menu-list-:rl:-menuitem-:rn:"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                 
                Delete
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
